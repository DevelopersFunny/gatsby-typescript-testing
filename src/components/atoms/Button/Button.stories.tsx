import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import { Button } from '.'

export default {
  title: 'Atoms/Button',
  component: Button,
  argTypes: {
    backgroundColor: { control: 'color' },
    color: { control: 'color' },
  },
} as ComponentMeta<typeof Button>

const Template: ComponentStory<typeof Button> = args => <Button {...args} />

export const Primary = Template.bind({})
Primary.args = {
  label: 'Primary Button',
  variant: 'primary',
}
Primary.parameters = {
  jest: ['button.test.tsx'],
}

export const Secondary = Template.bind({})
Secondary.args = {
  label: 'Secondary Button',
  variant: 'secondary',
}
Secondary.parameters = {
  jest: ['button.test.tsx'],
}

export const Small = Template.bind({})
Small.args = {
  label: 'Small Button',
  size: 'small',
  variant: 'primary',
}
Small.parameters = {
  jest: ['button.test.tsx'],
}

export const Large = Template.bind({})
Large.args = {
  label: 'Large Button',
  size: 'large',
  variant: 'primary',
}
Large.parameters = {
  jest: ['button.test.tsx'],
}

export const CustomBackgroundColor = Template.bind({})
CustomBackgroundColor.args = {
  backgroundColor: 'red',
  label: 'Custom Color',
  variant: 'primary',
}
CustomBackgroundColor.parameters = {
  jest: ['button.test.tsx'],
}

export const CustomTextColor = Template.bind({})
CustomTextColor.args = {
  color: 'gray',
  label: 'Custom Color',
  variant: 'primary',
}
CustomTextColor.parameters = {
  jest: ['button.test.tsx'],
}
