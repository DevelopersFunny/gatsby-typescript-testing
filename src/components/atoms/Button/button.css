.btn {
  @apply rounded-full focus:outline-none focus:ring text-sm px-3 py-2;
}

.text {
  @apply bg-transparent;
}
.contained {
  @apply text-white bg-blue-600 hover:bg-blue-700 active:bg-blue-600 focus:bg-blue-600/90;
}
.outlined {
  @apply border-2 border-blue-600 hover:border-blue-700 active:border-blue-600 focus:border-blue-600/90;
}

.small {
  @apply text-sm px-2 py-1;
}
.normal {
  @apply text-base px-3 py-2;
}
.large {
  @apply text-lg px-4 py-3;
}

.secondary {
  @apply text-white bg-gray-600 hover:bg-gray-700 active:bg-gray-600 focus:bg-gray-600/90 ring-gray-700;
}
