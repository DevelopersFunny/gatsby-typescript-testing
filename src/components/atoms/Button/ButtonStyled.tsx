import tw, { styled } from 'twin.macro'
import { IButtonProps } from '../../../types/components/button'

export const StyledButton = styled.button(({ variant, size }: IButtonProps) => [
  tw`
    rounded-full
    focus:outline-none
    focus:ring
  `,
  variant === 'contained'
    ? tw`text-white bg-blue-600 hover:bg-blue-700 active:bg-blue-600 focus:bg-blue-600/90`
    : variant === 'outlined'
    ? tw`border-2 border-blue-600 hover:border-blue-700 active:border-blue-600 focus:border-blue-600/90`
    : tw`bg-transparent`,
  size === 'small'
    ? tw`text-sm px-2 py-1`
    : size === 'large'
    ? tw`text-lg px-4 py-3`
    : tw`text-base px-3 py-2`,
])
