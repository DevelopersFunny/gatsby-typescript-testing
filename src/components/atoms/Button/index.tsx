import React from 'react'
import { IButtonProps } from '../../../types/components/button'
import { StyledButton } from './ButtonStyled'

export const Button = ({
  variant,
  size,
  backgroundColor,
  color,
  textColor,
  onClick,
  children,
  disabled,
}: IButtonProps) => {
  return (
    <StyledButton
      variant={variant}
      size={size}
      className={color}
      style={{ backgroundColor, color: textColor }}
      onClick={onClick}
      disabled={disabled}
    >
      {children}
    </StyledButton>
  )
}
