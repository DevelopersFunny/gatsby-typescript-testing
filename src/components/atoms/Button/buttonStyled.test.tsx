import React from 'react'
import * as stories from './Button.stories'
import { composeStories } from '@storybook/testing-react'
import { render, screen } from '@testing-library/react'
import { createSerializer, matchers } from '@emotion/jest'

/**
 * Emotion matchers and snapshots
 */
expect.extend(matchers)
expect.addSnapshotSerializer(createSerializer())

/**
 * Stories render
 */
const { Contained } = composeStories(stories)

/**
 * Tests
 */
describe('Button render with props', () => {
  it('should render style contained button', () => {
    render(<Contained />)
    const ButtonElement = screen.getByRole('button')
    expect(ButtonElement).toMatchSnapshot()
    expect(ButtonElement).toHaveStyleRule(
      'background-color',
      'rgba(37, 99, 235, var(--tw-bg-opacity))'
    )
  })
})
