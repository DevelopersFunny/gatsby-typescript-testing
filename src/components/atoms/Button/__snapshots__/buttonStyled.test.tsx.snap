// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with props should render style contained button 1`] = `
.emotion-0 {
  border-radius: 9999px;
  --tw-text-opacity: 1;
  color: rgba(255, 255, 255, var(--tw-text-opacity));
  --tw-bg-opacity: 1;
  background-color: rgba(37, 99, 235, var(--tw-bg-opacity));
  font-size: 1rem;
  line-height: 1.5rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.emotion-0:focus {
  outline: 2px solid transparent;
  outline-offset: 2px;
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(3px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow, 0 0 #0000);
}

.emotion-0:hover {
  --tw-bg-opacity: 1;
  background-color: rgba(29, 78, 216, var(--tw-bg-opacity));
}

.emotion-0:active {
  --tw-bg-opacity: 1;
  background-color: rgba(37, 99, 235, var(--tw-bg-opacity));
}

.emotion-0:focus {
  background-color: rgba(37, 99, 235, 0.9);
}

.emotion-0 {
  border-radius: 9999px;
  --tw-text-opacity: 1;
  color: rgba(255, 255, 255, var(--tw-text-opacity));
  --tw-bg-opacity: 1;
  background-color: rgba(37, 99, 235, var(--tw-bg-opacity));
  font-size: 1rem;
  line-height: 1.5rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.emotion-0:focus {
  outline: 2px solid transparent;
  outline-offset: 2px;
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(3px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow, 0 0 #0000);
}

.emotion-0:hover {
  --tw-bg-opacity: 1;
  background-color: rgba(29, 78, 216, var(--tw-bg-opacity));
}

.emotion-0:active {
  --tw-bg-opacity: 1;
  background-color: rgba(37, 99, 235, var(--tw-bg-opacity));
}

.emotion-0:focus {
  background-color: rgba(37, 99, 235, 0.9);
}

<button
  class="emotion-0"
>
  Contained Button
</button>
`;
